jetty-util/src/main/java/org/eclipse/jetty/util/BlockingCallback.java

```diff
@@ -0,0 +1,84 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2020 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under
+// the terms of the Eclipse Public License 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0
+//
+// This Source Code may also be made available under the following
+// Secondary Licenses when the conditions for such availability set
+// forth in the Eclipse Public License, v. 2.0 are satisfied:
+// the Apache License v2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.util;
+
+import java.io.IOException;
+import java.io.InterruptedIOException;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+public class BlockingCallback implements Callback
+{
+    private FutureCallback callback = new FutureCallback();
+    private final long timeout;
+
+    public BlockingCallback()
+    {
+        this(-1);
+    }
+
+    public BlockingCallback(long timeout)
+    {
+        this.timeout = timeout;
+    }
+
+    @Override
+    public void succeeded()
+    {
+        callback.succeeded();
+    }
+
+    @Override
+    public void failed(Throwable x)
+    {
+        callback.failed(x);
+    }
+
+    public void block() throws IOException
+    {
+        try
+        {
+            if (timeout > 0)
+                callback.get(timeout, TimeUnit.MILLISECONDS);
+            else
+                callback.get();
```

Comment:
This infinite timeout bothers me.
I have flashbacks to #3279 and #2061

Reply:
This is a jetty-util class so it may be used for another scenario where the infinite timeout makes sense.